app/api/V1/views/product.py

```diff
@@ -17,32 +17,36 @@
 
 parser = reqparse.RequestParser()
 
-parser.add_argument('product_name')
-parser.add_argument('category')
-parser.add_argument('description')
-parser.add_argument('inventory')
-parser.add_argument('price')
+parser.add_argument('product_name', required=True)
+parser.add_argument('category', required=True)
+parser.add_argument('description', required=True)
+parser.add_argument('inventory', required=True)
+parser.add_argument('price', required=True)
     
 
 
 @ns_product.route('')
 class ProductEndpoint(Resource):
     """Contains all the endpoints for Product Model"""
+    
 
     @ns_product.expect(product_models)
     def post(self):
-      
-
-
-
       args = parser.parse_args()
-      product_name = args['product_name']
+      product_names = args['product_name']
+      for product_name in product_names:
```

Comment:
Is product_names a list? If it is not, then you need to get rid of the for statement

Reply:
what does this product_names contain?

Reply:
its suppose to be a string that allows you to input a name of a product during post

Reply:
just below the product_names, include print(product_names), run the app with post request and see what is printed on the terminal

Reply:
it returns an empty field

Reply:
the research i have done while using restplus  when i initialiaze the field to be true it shouldnt allow a user to make a post without filling in the field
![capture](https://user-images.githubusercontent.com/33416672/47344375-fa7ec400-d6b0-11e8-9d65-29fa74e71ab3.JPG)

Reply:
use if statement not for

Reply:
eg. `if not product_names:`

Reply:
thanks that worked